server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -117,26 +120,35 @@ public SeLionRemoteProxy(RegistrationRequest request, Registry registry) throws
         proxyLogger.info(info.toString());
         info = new StringBuffer();
         info.append("SeLionRemoteProxy will attempt to recycle the node [");
-        info.append(machine).append("] after ").append(maxSessionsAllowed).append(" unique sessions");
+        info.append(machine).append("] after ").append(getMaxSessionsAllowed()).append(" unique sessions");
         proxyLogger.info(info.toString());
+
+        // detect presence of SeLion servlet capabilities on proxy
+        canForceShutdown = isSupportedOnNode(NodeForceRestartServlet.class);
+        canAutoUpgrade = isSupportedOnNode(NodeAutoUpgradeServlet.class);
+        canViewLogs = isSupportedOnNode(LogServlet.class);
+
+        // push these important values to the registered configuration object.
+        getConfig().put("uniqueSessionCount", getMaxSessionsAllowed());
+        getConfig().put("nodeRecycleThreadWaitTimeout", getNodeRecycleThread().getThreadWaitTimeout());
     }
 
     private boolean isSupportedOnNode(Class<? extends HttpServlet> servlet) {
         LOGGER.entering();
 
-        RequestConfig config = RequestConfig.custom().setConnectTimeout(CONNECTION_TIMEOUT).setSocketTimeout(CONNECTION_TIMEOUT).build();
+        RequestConfig config = RequestConfig.custom().setConnectTimeout(CONNECTION_TIMEOUT)
+                .setSocketTimeout(CONNECTION_TIMEOUT).build();
 
         CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
-        String url = String.format("http://%s:%d/extra/%s", machine, getRemoteHost().getPort(),
-                servlet.getSimpleName());
+        String url = String
+                .format("http://%s:%d/extra/%s", machine, getRemoteHost().getPort(), servlet.getSimpleName());
```

Comment:
ugh. dislike that the formatter inserted the line break here at the `.` instead of on a method param.. will fix.